src/org/rust/lang/core/psi/impl/RustNamedElementImpl.kt

```diff
@@ -19,7 +19,7 @@ public abstract class RustNamedElementImpl(node: ASTNode)   : RustCompositeEleme
         throw UnsupportedOperationException();
     }
 
-    override fun getNavigationElement() = getNameElement()
+    override fun getNavigationElement() = getNameElement() ?: this
```

Comment:
Hm, it is even more surprising: `getNavigationElement` in `PsiElementBase` is `@NotNull`, so `PsiElement` is the correct type here.